bundles/org.openhab.binding.smartthings/src/main/java/org/openhab/binding/smartthings/internal/SmartthingsHandlerFactory.java

```diff
@@ -0,0 +1,235 @@
+    }
+
+    public SmartthingsHandlerFactory() {
+        // Get a Gson instance
+        gson = new Gson();
+        // testing by bob
+        ChannelTypeRegistry ctr = getChannelTypeRegistry();
+    }
+
+    @Override
+    @Activate
+    public void activate(org.osgi.service.component.ComponentContext componentContext) {
+        super.activate(componentContext);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+        logger.debug("SmartthingsHandlerFactory is now processing ThingTypeUID {}", thingTypeUID.getAsString());
+
+        if (thingTypeUID.equals(THING_TYPE_SMARTTHINGS)) {
+            bridgeHandler = new SmartthingsBridgeHandler((Bridge) thing, this, bundleContext);
+            return bridgeHandler;
+        } else if (SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID)) {
+            // Everything but the bridge is handled by this one handler
+            SmartthingsThingHandler thingHandler = new SmartthingsThingHandler(thing, this);
+            thingHandlers.add(thingHandler);
+            return thingHandler;
+        }
+        return null;
+    }
+
+    /**
+     * Remove handler of things.
+     */
+
+    @Override
+    protected synchronized void removeHandler(ThingHandler thingHandler) {
+        if (thingHandler instanceof SmartthingsBridgeHandler) {
+            ServiceRegistration<?> serviceReg = this.discoveryServiceRegs.get(thingHandler.getThing().getUID());
+            serviceReg.unregister();
+            discoveryServiceRegs.remove(thingHandler.getThing().getUID());
+        }
+    }
+
+    /**
+     * Send a command to the Smartthings Hub
+     *
+     * @param path http path which tells Smartthings what to execute
+     * @param data data to send
+     * @return Response from Smartthings
+     * @throws InterruptedException
+     * @throws TimeoutException
+     * @throws ExecutionException
+     */
+    public @Nullable Map<String, Object> sendDeviceCommand(String path, String data)
+            throws InterruptedException, TimeoutException, ExecutionException {
+        ContentResponse response = httpClient
+                .newRequest(bridgeHandler.getSmartthingsIp(), bridgeHandler.getSmartthingsPort())
+                .timeout(3, TimeUnit.SECONDS).path(path).method(HttpMethod.POST)
+                .content(new StringContentProvider(data), "application/json").send();
+
+        Map<String, Object> result = null;
+        int status = response.getStatus();
+        if (status == 200) {
+            String responseStr = response.getContentAsString();
+            if (responseStr != null && responseStr.length() > 0) {
+                result = new HashMap<String, Object>();
+                result = gson.fromJson(responseStr, result.getClass());
+            }
+        } else if (status == 202) {
+            logger.info(
```

Comment:
I removed the code related to status == 200 as it will never happen. 202 is the normal response code. The actual response message is sent by Smartthings as a separate message.